### PR TITLE
LDAP: Compute values when reloading LDAP settings

### PR DESCRIPTION
**Pull request by dmihai**
**What is this feature?**

Calculate `MinTLSVersionID` and `TLSCipherIDs` from input text values when reloading LDAP settings. Also initialize `Timeout` and `OrgID` from group mappings with default values if they are not configured.

**Why do we need this feature?**

For LDAP configurations.

**Who is this feature for?**

Users using LDAP via SSO settings.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by volcanonoodle — APPROVED** [review]

File: pkg/services/ldap/service/ldap.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 	"sync"
 
 	"github.com/grafana/grafana/pkg/apimachinery/identity"
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/logger"
```
Comment (mgyongyosi) [inline pkg/services/ldap/service/ldap.go:10]: Could you please replace this with `github.com/grafana/grafana/pkg/infra/log`?

**Review by mgyongyosi — APPROVED** [review]

File: pkg/services/ldap/service/ldap.go (hunk 1 above)
Comment (dmihai) [reply pkg/services/ldap/service/ldap.go]: Good catch, I fixed it.

File: pkg/services/ldap/service/ldap.go [hunk 2]
```diff
@@ -99,6 +99,34 @@ func (s *LDAPImpl) Reload(ctx context.Context, settings models.SSOSettings) erro
 		return err
 	}
 
+	// calculate MinTLSVersionID and TLSCipherIDs from input text values
+	// also initialize Timeout and OrgID from group mappings with default values if they are not configured
+	for _, server := range ldapCfg.Servers {
+		if server.MinTLSVersion != "" {
+			server.MinTLSVersionID, err = util.TlsNameToVersion(server.MinTLSVersion)
+			if err != nil {
+				s.log.Error("failed to set min TLS version, ignoring", "err", err)
```
Comment (linoman) [inline pkg/services/ldap/service/ldap.go:108]: i'd add the server host to the output for a more detailed message

File: pkg/services/ldap/service/ldap.go [hunk 3]
```diff
@@ -99,6 +99,34 @@ func (s *LDAPImpl) Reload(ctx context.Context, settings models.SSOSettings) erro
...
+			}
+		}
+
+		if len(server.TLSCiphers) > 0 {
+			server.TLSCipherIDs, err = util.TlsCiphersToIDs(server.TLSCiphers)
+			if err != nil {
+				s.log.Error("unrecognized TLS Cipher(s), ignoring", "err", err)
```
Comment (linoman) [inline pkg/services/ldap/service/ldap.go:115]: same here

**Review by linoman — APPROVED** [review]
Great work! I added some comments to improve the verbosity of the logs. It is not a blocker but it might be nice to include them.

File: pkg/services/ldap/service/ldap.go (hunk 2 above)
Comment (dmihai) [reply pkg/services/ldap/service/ldap.go]: Good one, thanks. I added it now.

**Pull request merged**